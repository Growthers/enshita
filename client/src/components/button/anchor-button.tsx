import type { FC } from "react";
import { InlineIcon } from "@iconify/react";
import styles from "~/styles/components/button/anchor-button.module.scss";
import type {
  ButtonIconProperties,
  AnchorButtonProperties,
} from "./type/model";

const ButtonIcon: FC<ButtonIconProperties> = ({ icon, css, ...rest }) => (
  <div className={`${styles["enshita-button-icon-box"]} ${css}`}>
    <InlineIcon
      icon={icon}
      className={styles["enshita-button-icon"]}
      {...rest}
    />
  </div>
);

const AnchorButton: FC<AnchorButtonProperties> = ({
  children,
  icon,
  variant,
  iconStyles,
  boxStyles,
  textStyles,
  isSamePage,
  href,
  ...rest
}) => (
  <a
    className={`${styles["enshita-button-box"]} ${boxStyles}`}
    href={href}
    target={isSamePage ? "" : "_blank"}
    rel="noopener noreferrer"
    role="button"
    {...rest}
  >
    {variant === "icon" && <ButtonIcon icon={icon} css={iconStyles} />}
    {children && (
      <p className={`${styles["enshita-button-text"]} ${textStyles}`}>
        {children}
      </p>
    )}
  </a>
);

export { AnchorButton };
