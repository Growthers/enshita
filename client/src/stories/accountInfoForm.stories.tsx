// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
