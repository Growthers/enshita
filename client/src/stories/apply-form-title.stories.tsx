import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
