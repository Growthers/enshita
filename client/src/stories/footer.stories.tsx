// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
