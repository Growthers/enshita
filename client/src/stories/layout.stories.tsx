import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
