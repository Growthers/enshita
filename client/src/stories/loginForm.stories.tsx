// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
