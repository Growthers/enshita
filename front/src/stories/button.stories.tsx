import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
