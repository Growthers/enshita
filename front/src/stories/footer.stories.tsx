import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
