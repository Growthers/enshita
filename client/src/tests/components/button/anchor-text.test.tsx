import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/button/anchor-button.stories";

const { Default, Icon } = composeStories(stories);

describe("(components) anchor button", () => {
  const options = { name: "This is test" };
  test("[role=button]である", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("button", options)).toBeInTheDocument();
  });
  test("(normal) take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("(icon) take snapshot", () => {
    const { container } = render(<Icon />);
    expect(container).toMatchSnapshot();
  });
});
