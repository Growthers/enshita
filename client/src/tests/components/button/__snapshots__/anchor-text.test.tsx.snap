// Vitest Snapshot v1

exports[`(components) anchor button > (icon) take snapshot 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 undefined"
    href="#"
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <div
      class="_enshita-button-icon-box_8w723_1 undefined"
    >
      <span />
    </div>
    <p
      class="_enshita-button-text_8w723_36 undefined"
    >
      This is test
    </p>
  </a>
</div>
`;

exports[`(components) anchor button > (normal) take snapshot 1`] = `
<div>
  <a
    class="_enshita-button-box_8w723_17 undefined"
    href="#"
    rel="noopener noreferrer"
    role="button"
    target="_blank"
  >
    <p
      class="_enshita-button-text_8w723_36 undefined"
    >
      This is test
    </p>
  </a>
</div>
`;
